Initial package setup (#1)
simpleclub/design_tokens_builder: README.md

<h1 align="center">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://github.com/simpleclub/design_tokens_builder/assets/35028202/0e70832e-78f2-4906-a5b1-940804f2619a">
      <img alt="Logo" src="https://github.com/simpleclub/design_tokens_builder/assets/35028202/6ec91bfa-3124-4430-9ed0-6d616fe5c838">
    </picture>

> </h1>

Reviewer: This looks bad when rendered on github

![design_tokens_builder on Pub](https://img.shields.io/pub/v/design_tokens_builder.svg)

A package for generating a flutter `ThemeData` from an exported design token json that is used e.g. [in this Figma plugin](https://tokens.studio/). It rather aims to create a flutter `Theme` 
based on Material 3 than creating a lot of custom widgets for each token.

## Features

Right now we support generating/parsing the following:
- `ColorScheme`
- `TextStyle`

>   - Font family mapping (from Figma font name to flutter name)

Reviewer 2: ```suggestion
  - Font family mapping (from Figma font name to Flutter name)
```

Reviewer 2: There are other occurrences of this typo.

  child: Container(),
);
```
By listening on Brightness changes you could then easily switch between themes. You could also easily change the token set. 

This package also exposes the generated theme extensions by building a `BuildContext` extension. You can use the shortcut like this `context.yourExtension`. We also provide shortcuts for theme related properties like `context.colorScheme` and `context.textTheme`.

## Additional information

### Multi-theming and dark and light mode
The package can generate multiple themes based on multiple [token sets](https://docs.tokens.studio/themes/token-sets) in
your token data json. Every set will be available through a `GeneratedTokenSet`. This enum contains all set pairs and 
`ThemeData` for light and dark themes. In order to specify the brightness of your set use Light/Dark as a suffix in 
design tokens (e.g. yourSetLight, yourSetDark). You also might have a global or core set with universal tokens that do 
not change based on the selected brightness. In that case make sure to call this set `global`. It will be recognised by 
the builder and ignored when creating `GeneratedTokenSets`.

### Generation of flutter `ThemeData`

> The package allows for easy and simple generation of flutters `ThemeData`. For doing so, you have to have a certain 

Reviewer 3: ```suggestion
The package allows for easy and simple generation of Flutter's `ThemeData`. For doing so, you have to have a certain 

```

structure for your design tokens. Since this is a system token related to material 3 the tokens used to generate flutters

> `ThemeData` should always start with `sys`. If you then e.g. want to set the primary color in flutters `ColorScheme` you

Reviewer 3: Any change the structure and the sys rule can be demonstrated with a code snippet or a screenshot?

have to add a token called `sys.primary` with type `color` to your json file containing all of your tokens. `primary` is
the name of `ColorScheme`s `primary` field. Keep in mind to exactly write the field names in camel case so the package is
able to recognize them properly. 
For text styles it works similar. Here also use `sys` as the leading part of your token. To better organize the tokens in
Token Studio we decided to split the text style naming. So if you want to generate `displaySmall` text style just use a
`sys.display.small` token of type `typography`.

### Generation of theme extensions

> Theme extension are a great way providing custom tokens that are not in flutters `ThemeData` defined. In order to create

Reviewer 3: ```suggestion
Theme extensions are a great way of providing custom parameters that are not defined in `ThemeData`. In order to create

```

a new extension you can just use a different prefix then `sys` for your tokens. Based on that name and the type of the

> tokens inside that group an extension will get generated and added to the `ThemeData`. So a custom set of colors called

Reviewer 3: > inside that group

Could you specify which group?
